modules/git/repo.go

```diff
@@ -209,49 +209,22 @@ func Push(ctx context.Context, repoPath string, opts PushOptions) error {
 	} else {
 		cmd.SetDescription(fmt.Sprintf("push branch %s to %s (force: %t, mirror: %t)", opts.Branch, opts.Remote, opts.Force, opts.Mirror))
 	}
-	var outbuf, errbuf strings.Builder
 
-	if opts.Timeout == 0 {
-		opts.Timeout = -1
-	}
```

Comment:
I was worried about removing this, but it seems OK.

https://github.com/go-gitea/gitea/blob/6bc3079c0036a54d1b8ab04c5a6e14111c719c9a/modules/git/command.go#L248

Reply:
Yup, this code ` if timeout <= 0 { ` was also written by me 🤣